app/src/main/java/org/kiwix/kiwixmobile/main/KiwixMainActivity.kt

```diff
@@ -179,10 +188,32 @@ class KiwixMainActivity : CoreMainActivity() {
 
   override fun onNewIntent(intent: Intent) {
     super.onNewIntent(intent)
+    handleNotificationIntent(intent)
     supportFragmentManager.fragments.filterIsInstance<FragmentActivityExtensions>().forEach {
       it.onNewIntent(intent, this)
     }
   }
 
+  @Suppress("MagicNumber")
+  private fun handleNotificationIntent(intent: Intent) {
+    if (intent.hasExtra(DOWNLOAD_NOTIFICATION_TITLE)) {
+      Handler(Looper.getMainLooper()).postDelayed(
+        {
+          intent.getStringExtra(DOWNLOAD_NOTIFICATION_TITLE)?.let {
+            newBookDao.bookMatching(it)?.let { bookOnDiskEntity ->
+              navigate(
```

Comment:
Please create a function for this for better readability of code like we have other functions `openSearch`, `openPage` etc so that it can be reused from other places as well. `openZimFromFilepath` would be better.

Reply:
Sounds good. I should make it public,right?

Reply:
Yes.

Reply:
I am getting an error "Function 'openZimFromFilePath' could be private", how to suppress?

Reply:
Resolved